Code/GraphMol/Depictor/DepictUtils.cpp

```diff
@@ -193,6 +193,88 @@ int pickFirstRingToEmbed(const RDKit::ROMol &mol,
   return res;
 }
 
+RDKit::VECT_INT_VECT findCoreRings(const RDKit::VECT_INT_VECT &fusedRings,
+                                   RDKit::INT_VECT &coreRingsIds) {
+  // simplify the fused rings to a set of core rings by iteratively removing
+  // rings that share only one or two consecutive atoms. These
+  // are trivial to embed after the core rings have been embedded and will make
+  // template matching more powerful since it will not be affected by the side
+  // rings
+
+  RDKit::INT_VECT removedRings;
+  bool removedARing = false;
+  do {
+    removedARing = false;
+    for (unsigned int currRingId = 0; currRingId < fusedRings.size();
+         currRingId++) {
+      if (std::find(removedRings.begin(), removedRings.end(), currRingId) !=
+              removedRings.end() ||
+          removedARing) {
+        continue;
+      }
+      std::set<int> allIntersectingAtoms;
+      RDKit::INT_VECT commmonAtoms;
+      for (unsigned int otherRingId = 0; otherRingId < fusedRings.size();
+           otherRingId++) {
+        if (currRingId == otherRingId ||
+            (std::find(removedRings.begin(), removedRings.end(), otherRingId) !=
+             removedRings.end())) {
+          continue;
+        }
+        RDKit::Intersect(fusedRings[currRingId], fusedRings[otherRingId],
+                         commmonAtoms);
+        if (commmonAtoms.size() > 0) {
+          for (auto rii : commmonAtoms) {
+            allIntersectingAtoms.insert(rii);
+          }
+        }
+      }
+      // note that the set of ring is not SSSR because we use symmetrizeSSSR, so
+      // we cannot force a check for only one fused ring. Instead we make sure
+      // that this ring shares only one atom or one bond (two consecutive atoms)
+      auto hasOneOrTwoConsecutiveAtoms = [fusedRings, currRingId](
+                                             const RDKit::INT_VECT &vec) {
+        if (vec.size() == 1) {
+          return true;
+        }
+        if (vec.size() == 2) {
+          auto ring = fusedRings[currRingId];
+          auto pos1 = find(ring.begin(), ring.end(), vec[0]);
+          auto pos2 = find(ring.begin(), ring.end(), vec[1]);
+
+          if (pos1 == ring.end() || pos2 == ring.end()) {
+            return false;
+          }
+          int pos1I = pos1 - ring.begin();
+          int pos2I = pos2 - ring.begin();
+          // check if the positions in the ring vector of the two
+          // elements are consecutive
+          return (abs(pos1I - pos2I) == 1 ||
+                  (pos1I == 0 && pos2I == static_cast<int>(ring.size()) - 1) ||
+                  (pos2I == 0 && pos1I == static_cast<int>(ring.size()) - 1));
```

Comment:
I think it would be ok to pass `dp_mol` into this function and just check if the atoms are bonded here